src/main/java/hvac/weather/parsing/WeatherSaver.java

```diff
@@ -0,0 +1,102 @@
+        try(BufferedReader reader = new BufferedReader(new FileReader(dataFile));
+            Connection connection = new Connection()) {
+            System.out.println("Parsing weather data");
+            WeatherData weatherData = gson.fromJson(reader, WeatherData.class);
+            System.out.println("Processing weather data");
+            WeatherSnapshot[] snapshots = processWeatherData(weatherData);
+            System.out.println("Saving to db");
+            saveSnapshots(snapshots, connection);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private static void saveSnapshots(WeatherSnapshot[] snapshots, Connection connection) {
+        EntityManager entityManager = connection.createEntityManager();
+        entityManager.getTransaction().begin();
+        String hql = "delete from WeatherSnapshot";
+        Query query = entityManager.createQuery(hql);
+        query.executeUpdate();
+        for (WeatherSnapshot snapshot: snapshots) {
+            entityManager.persist(snapshot);
+        }
+        entityManager.getTransaction().commit();
+        entityManager.close();
+    }
+
+    private static File getFileFromResources() {
+        ClassLoader classLoader = WeatherSaver.class.getClassLoader();
+
+        URL resource = classLoader.getResource(WeatherSaver.weatherDataResource);
+        if (resource == null) {
+            throw new IllegalArgumentException("file is not found!");
+        } else {
+            return new File(resource.getFile());
+        }
+    }
+    private static WeatherSnapshot[] processWeatherData(WeatherData weatherData) {
+        WeatherSnapshot[] result = new WeatherSnapshot[weatherData.data.length];
+        for(int i = 0; i < result.length; i++) {
+            result[i] = new WeatherSnapshot();
+            result[i].setPressureHPa(weatherData.data[i].pressure);
+            result[i].setTemperatureKelvin(weatherData.data[i].temperature + 273);
+            result[i].setDate(weatherData.data[i].time_local);
+            result[i].setAbsoluteHumidity(calculateAbsoluteHumidity(weatherData.data[i]));
+        }
+        return result;
+    }
+
+    private static float calculateAbsoluteHumidity(WeatherData.WeatherEntry weatherEntry) {
+        //https://planetcalc.com/2161/
+        double pressureFunctionValue = 1.0016
+                + 0.00000315 * weatherEntry.pressure
+                - 0.074 / weatherEntry.pressure;
+        float saturationWaterVapour = (float)(pressureFunctionValue * 6.112
+                * Math.exp(17.62*weatherEntry.temperature/(243.12+weatherEntry.temperature)));
+        //https://planetcalc.com/2167/
+        float currentWaterVapour = saturationWaterVapour * weatherEntry.humidity;
+        float result = currentWaterVapour/(461.5f*(weatherEntry.temperature+273));
+        return currentWaterVapour/(461.5f*(weatherEntry.temperature+273));
```

Comment:
Tu zamiast "273" (a może nawet innych) przydałoby się wprowadzić stałe, a nawet cały plik ze stałymi fizycznymi i współczynnikami, żeby wszędzie uzywać tych samych liczb, inaczej np. raz napiszemy -273, innym razem -273.15, i będziemy mieli niespójności.

Reply:
To może w oddzielny PR to naprawię, bo pewnie też tak trzeba zrobić w SimulationAgent